proposal/custom-url-resolution.md

```diff
@@ -0,0 +1,117 @@
+# Custom url resolution: Draft 1
+
+_[(Issue)](https://github.com/sass/sass/issues/2535)_
+
+## Table of Contents
+
+- [Background](#background)
+- [Summary](#summary)
+  - [Steps](#steps)
+  - [JavaScript API](#javaScript-api)
+  - [Edge cases](#edge-cases)
+- [Syntax](#syntax)
+- [Deprecation Process](#deprecation-process)
+
+## Background
+
+> This section is non-normative.
+
+Many css features require the use of a url import to reference resources from outside the sass files, however these files also need to exist on the eventual output directory and server. To ensure the references are valid, the sass API should allow for the user to provide a way to remap and/or inline these resources.
+
+## Summary
+
+> This section is non-normative.
+
+This proposal defines a standardized way to remap the url imports to the final location on the server or inline reference.
+
+This is accomplished by running url references through the importer plugin(s) and url rewriting plugin(s) if any url rewriting plugin has been defined.
+
+### Steps
+
+The steps of a url reference import:
+
+- Url reference gets extracted from the Sass file (For more information see [Syntax](#syntax))
+- This reference gets remapped to the actual location on the filesystem by the importer plugin(s)
+- The resolved reference gets passed into the rewrite url plugin pipeline which returns the output url of this resource.
+
+If a certain resource cannot be remapped the url rewrite plugin will return `null`, if all rewrite plugins return `null` it should fallback to the original value of the url reference.
```

Comment:
What does a `null` return value mean? Does it mean "I don't understand this URL format"? Does it mean "I couldn't find the file this URL was referring to"? We need to be very clear about these semantics because (if we do allow multiple rewriters) it's entirely possible that multiple URL rewriters will coexist in the same environment, and they need to have a shared understanding of how they're expected to operate in order to cooperate without having direct knowledge of one another.

Reply:
It meant `I don't understand this URL format` but on second thought this is probably not really gonna happen, so removed this ability and the multiple plugin pipeline support.